### DISCUSSION TITLE
How to integrate in Astrojs project

### DISCUSSION THREAD

**Discussion by princemuel** [edmundhung/conform#378] [Q&A]
Hi there :wave: Please, I came across this package recently and tried it out in one of my Remix projects and it was perfect. 

I've been having issues though integrating it in my Astrojs project and need help with validating a form (contact form) server-side and passing the errors to the client (No JS Validation). 

Thanks

**Comment by princemuel** [comment]
Oh! :hand_over_mouth:  I just noticed the package is specific to Remix Run and React Router. Still I feel like there should be a way to still integrate it

**Comment by edmundhung** [comment]
You can use it without Remix or React Router. Conform depends only on the react package at the moment.

I am not familiar with Astro. But here is the general idea:

```tsx
export function ExampleForm() {
  const [submission, setSubmission] = useState();
  const [form, fields] = useForm({
    lastSubmission: submission,
    async onSubmit(event, { action, method, submission }) {
      event.preventDefault();

      // Call your action handler
      const response = await fetch(action, {
        method,
        body: JSON.stringify(submission.value),
      });
      const result = await response.json();

      // Assume you are returning `status` and the `submission` object 
      if (result.status === 'error') {
        setSubmission(result.submission);
        return;
      }

      // ... do something else
    },
  });

  // ...
}
```

**Reply by princemuel** [reply]
I kinda wanted to implement Progressive Enhancement on the form. Right now, It's working and I'm submitting but the values in the form get wiped out when there's an error.
